pilot/cmd/pilot-agent/main.go

```diff
@@ -476,13 +479,19 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
-// check if SDS UDS path and token path exist, if both exist, requests key/cert
-// using SDS instead of secret mount.
+// detectSds checks config for control plane SDS. The returned values are used by caller
+// to decide whether SDS is used for control plane.
 func detectSds(controlPlaneBootstrap, controlPlaneAuthEnabled bool, udspath, trustworthyJWTPath string) (bool, string) {
 	if !sdsEnabledVar.Get() {
 		return false, ""
 	}
 
+	// If controlplanesecurity is disabled, return false so that Pilot agent does not create SDS config
+	// in bootstrap config.
+	if !controlPlaneAuthEnabled {
```

Comment:
Doesn't make any sense. This detects sds, has nothing to do with controlplane auth policy,

Reply:
To be clear: the method itself is badly named and designed. "detectSds" should detect if SDS socket is present. It doesn't matter if control plane auth is enabled - sds can be used for other things.

Reply:
I want to make it clear that this is a "bug fix" not a code refactoring. Can we not conflate all the changes in this single PR? Our goal is simple: make the minimal necessary fixes to make the control plane SDS working in 1.3. Other work seems not related to this PR.
@duderino